aten/src/ATen/native/cuda/LossCTC.cu

```diff
@@ -26,9 +26,15 @@ namespace {
 
 // this ad-hoc converts from targets (l in [1]) to augmented targets (l' in [1]) note that no bound-checking is done
 // __restrict__ impact to be measured, https://devblogs.nvidia.com/cuda-pro-tip-optimize-pointer-aliasing/
-template<typename target_t>
-__device__ static inline int64_t get_target_prime(const target_t* __restrict__ target, int64_t offset, int64_t stride, int64_t idx, int64_t BLANK) {
-  if (idx % 2 == 0) {
+template <typename target_t>
+__device__ static inline int64_t get_target_prime(
```

Comment:
This is me being ignorant about how CTCLoss works :) For my education, what exactly does `get_target_prime` do? E.g., what is it called in the paper?

Reply:
You've adjusted this function to handle `target_length == 0` but in many of the call sites, there is already a condition that implies that if you get to this function, target length is nonzero. Does this hold in all of the call sites? I guess I'll go check now.

Reply:
I guess in backwards, there are a few cases when you will get here when `target_length == 0`.

Reply:
So the comment above the function `// this ad-hoc converts from targets (l in [1]) to augmented targets (l' in [1]) note that no bound-checking is done` isn't all that great (when l is the targets, l' is `BLANK l_0 BLANK l_1 ... l_targetlen BLANK`.

I'll do a bit more analysis if we need the target length condition here. It might well be that it is not called except with idx 0, which would be equally well...